### PR TITLE
feat: Refactored Auth feature to Riverpod and MVVM

### LINKED ISSUE #817: Refactor Auth Module from GetX + MVC to Feature-Based MVVM with Riverpod
### 🚀 The feature

Refactor the existing authentication module from the current MVC architecture using GetX to a feature-based MVVM architecture powered by Riverpod. 

### Motivation, pitch

The current implementation tightly couples business logic, state management, and UI concerns, making long-term maintenance and feature expansion more challenging. Migrating to a feature-based MVVM structure with Riverpod will provide clearer separation of concerns, better dependency management, improved testability, and a more scalable foundation for future development.

### PR DESCRIPTION
**Pull request by Mayank4352**
## Description

Refactor the existing authentication module from the current MVC architecture using GetX to a feature-based MVVM architecture powered by Riverpod.

Fixes #817 

## Type of change

- [x] Refactor (does not change functionality, e.g. code style improvements, linting)

## How Has This Been Tested?

Tested Locally

Please include screenshots below if applicable.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #817 
- [ ] Tag the PR with the appropriate labels

### CODE REVIEW

**Comment by Mayank4352** [comment]
Hey @M4dhav, Kindly review the PR

**Comment by Mayank4352** [comment]
image
Hey @M4dhav, as mentioned in #816 tests are failing, due to the missing dependency. But they are passing locally

File: lib/main.dart [hunk 1]
```diff
@@ -68,19 +71,29 @@ Future<void> main() async {
...
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
     UiSizes.init(context);
     final themeController = Get.put(ThemeController());
+    final router = ref.watch(routerProvider);
 
     return Obx(
-      () => GetMaterialApp(
+      () => GetMaterialApp.router(
```
Comment (4555jan) [inline lib/main.dart:92]: This still uses GetMaterialApp instead of a standard MaterialApp. Since we're migrating to Riverpod, this should be wrapped with ProviderScope  isnt?

File: lib/main.dart [hunk 2]
```diff
@@ -59,6 +61,7 @@ Future<void> main() async {
   await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
   await GetStorage.init();
   Get.put(AboutAppScreenController());
```
Comment (4555jan) [inline lib/main.dart:63]: Get.put(AboutAppScreenController()) is still registering dependencies via GetX's  system, while the new code uses Riverpod's UncontrolledProviderScope with a Container in runApp(). These are two completely separate dependency registries isnt

File: lib/main.dart [hunk 3]
```diff
@@ -59,6 +61,7 @@ Future<void> main() async {
...
+  setupLegacyGetXDependencies();
```
Comment (4555jan) [inline lib/main.dart:64]: The presence of setupLegacyGetXDependencies() . The function name explicitly says these are legacy dependencies, yet they're still being called in main() after the Riverpod UncontrolledProviderScope setup.

**Review by 4555jan — CHANGES_REQUESTED** [review]
in main.dart file  GetX and Riverpod are being used simultaneously. please resolve this

File: lib/main.dart (hunk 1 above)
Comment (Mayank4352) [reply lib/main.dart:92]: Riverpod is already wired, UncontrolledProviderScope at line 74 sits above GetMaterialApp.router, they're independent. Kept GetMaterialApp because legacy controllers still call Get.snackbar, Get.dialog etc

File: lib/main.dart (hunk 2 above)
Comment (Mayank4352) [reply lib/main.dart:63]: That's intentional during a feature-by-feature migration only Auth is on Riverpod so far. The bridge in [core/container.dart] lets legacy GetX controllers read Riverpod auth state without being Riverpod-aware. AboutAppScreenController isn't auth-related

File: lib/main.dart (hunk 3 above)
Comment (Mayank4352) [reply lib/main.dart:64]: These are GetX controllers that haven't been migrated to Riverpod yet. Previously each route's Binding registered them on entry. GoRouter doesn't support bindings, so I register them all once at startup here. As each feature migrates to Riverpod, its line gets removed the whole function goes away when nothing is left on GetX.

**Comment by Mayank4352** [comment]
Hey @M4dhav, Kindly review the PR

File: lib/routes/app_router.dart [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+        ),
+      ),
+    ],
+  );
+});
+
+GoRouter get appRouter => rootContainer.read(routerProvider);
+
+String? _redirect(Ref ref, GoRouterState state) {
+  final auth = ref.read(authProvider).value;
```
Comment (Muneerali199) [inline lib/routes/app_router.dart]: Hey, nice work on the auth migration! One thing I noticed - when `loadCurrentUser()` throws (say network error), `authProvider.value` returns null and the redirect at line 176 returns null too. This means the user stays on a protected page with no auth data, which can crash the screen. Might need to handle `AsyncError` separately here.

File: lib/features/auth/data/auth_repository.dart [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+  // OTP / email verification
+
+  Future<({String otpId, String responseBody})> sendOtp({
+    required String email,
+  }) async {
+    var otpId = randomNumeric(10) + email;
+    otpId = otpId.split('@')[0];
+
+    await _account.updatePrefs(
+      prefs: {'otp_ID': otpId, 'isUserProfileComplete': true},
```
Comment (Muneerali199) [inline lib/features/auth/data/auth_repository.dart]: Small thing I noticed - `isUserProfileComplete: true` is set here during `sendOtp()`, but onboarding hasn't finished yet. If the app restarts after OTP is sent but before onboarding completes, `loadCurrentUser()` will see this as true and try to fetch a DB row that doesn't exist yet. Maybe move this to after onboarding actually completes?

File: lib/routes/route_paths.dart [hunk 6]
```diff
@@ -0,0 +1,86 @@
+class RoutePaths {
+  RoutePaths._();
+
+  // Auth flow
+  static const splash = '/splash';
+  static const landing = '/landing';
+  static const welcome = '/welcomeScreen';
+  static const login = '/loginScreen';
+  static const signup = '/signup';
+  static const emailVerification = '/emailVerification';
```
Comment (Muneerali199) [inline lib/routes/route_paths.dart:10]: Noticed that `emailVerification` is defined here but isn't in either `authOnly` or `protected` sets below. So both logged-in and logged-out users can access it freely — might not be what you intended.

Comment (Mayank4352) [reply lib/routes/route_paths.dart:10]: yep, mb should be in protected

File: lib/features/auth/data/auth_repository.dart (hunk 5 above)
Comment (Mayank4352) [reply lib/features/auth/data/auth_repository.dart]: yeah, older versions had it implemented in a similar way but this does makes an edge case here

File: lib/routes/app_router.dart (hunk 4 above)
Comment (Mayank4352) [reply lib/routes/app_router.dart]: yep, will add a test case for this
worth adding

**Review by M4dhav — CHANGES_REQUESTED** [review]
Please open another PR to fix the failing Test Workflow before this PR
